testing/tokenfilter/tokenfilter_test.go

```diff
@@ -0,0 +1,132 @@
+
+	// Default IBC Simapp
+	otherChain *ibctesting.TestChain
+}
+
+func (suite *TokenFilterTestSuite) SetupTest() {
+	chains := make(map[string]*ibctesting.TestChain)
+	suite.coordinator = &ibctesting.Coordinator{
+		T:           suite.T(),
+		CurrentTime: time.Now(),
+		Chains:      chains,
+	}
+	suite.celestiaChain = NewTestChain(suite.T(), suite.coordinator, ibctesting.GetChainID(1))
+	suite.otherChain = ibctesting.NewTestChain(suite.T(), suite.coordinator, ibctesting.GetChainID(2))
+
+	suite.coordinator.Chains[ibctesting.GetChainID(1)] = suite.celestiaChain
+	suite.coordinator.Chains[ibctesting.GetChainID(2)] = suite.otherChain
+}
+
+func NewTransferPath(celestiaChain, otherChain *ibctesting.TestChain) *ibctesting.Path {
+	path := ibctesting.NewPath(celestiaChain, otherChain)
+	path.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
+	path.EndpointB.ChannelConfig.PortID = ibctesting.TransferPort
+	path.EndpointA.ChannelConfig.Version = types.Version
+	path.EndpointB.ChannelConfig.Version = types.Version
+
+	return path
+}
+
+// TestHandleOutboundTransfer asserts that native tokens on a celestia based chain can be transferred to
+// another chain and can then return to the original celestia chain
+func (suite *TokenFilterTestSuite) TestHandleOutboundTransfer() {
+	// setup between celestiaChain and otherChain
+	path := NewTransferPath(suite.celestiaChain, suite.otherChain)
+	suite.coordinator.Setup(path)
+
+	celestiaApp := suite.celestiaChain.App.(*app.App)
+	originalBalance := celestiaApp.BankKeeper.GetBalance(suite.celestiaChain.GetContext(), suite.celestiaChain.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+	// take half of the original balance
+	amount := originalBalance.Amount.QuoRaw(2)
+	timeoutHeight := clienttypes.NewHeight(1, 110)
+	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, amount)
+
+	// send half the users balance from celestiaChain to otherChain
+	msg := types.NewMsgTransfer(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, coinToSendToB, suite.celestiaChain.SenderAccount.GetAddress().String(), suite.otherChain.SenderAccount.GetAddress().String(), timeoutHeight, 0, "")
+	res, err := suite.celestiaChain.SendMsgs(msg)
+	suite.Require().NoError(err) // message committed
+
+	packet, err := ibctesting.ParsePacketFromEvents(res.GetEvents())
+	suite.Require().NoError(err)
+
+	// relay send
+	err = path.RelayPacket(packet)
+	suite.Require().NoError(err) // relay committed
+
+	// check that the token exists on chain B
+	voucherDenomTrace := types.ParseDenomTrace(types.GetPrefixedDenom(packet.GetDestPort(), packet.GetDestChannel(), sdk.DefaultBondDenom))
+	balance := suite.otherChain.GetSimApp().BankKeeper.GetBalance(suite.otherChain.GetContext(), suite.otherChain.SenderAccount.GetAddress(), voucherDenomTrace.IBCDenom())
+	coinSentFromAToB := types.GetTransferCoin(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, sdk.DefaultBondDenom, amount)
+	suite.Require().Equal(coinSentFromAToB, balance)
+
```

Comment:
[optional]

```go
	// check that the account on celestiaChain has less tokens than the original starting balance
	celestiaBalance := celestiaApp.BankKeeper.GetBalance(suite.celestiaChain.GetContext(), suite.celestiaChain.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
	suite.Require().True(celestiaBalance.Amount.LT(originalBalance.Amount))
```

[question] Does this test intentionally use the `stake` denomination instead of `utia`?

```
celestiaBalance: 5000000000000000000stake
originalBalance: 10000000000000000000stake
```

 If not, proposal to use `utia`. On second thought, maybe it intentionally doesn't use `utia` to avoid depending on the `app` package.

Reply:
> suite.Require().True(celestiaBalance.Amount.LT(originalBalance.Amount))

I would prefer to be exact.

> [question] Does this test intentionally use the stake denomination instead of utia?

No intentions around the denomination. I just used whatever the defaults are

Reply:
> I would prefer to be exact.

An exact assertion also works so something like

```go
	// check that the account on celestiaChain has fewer tokens than it started with
	celestiaBalance := celestiaApp.BankKeeper.GetBalance(suite.celestiaChain.GetContext(), suite.celestiaChain.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
	want := originalBalance.Amount.Sub(coinToSendToB.Amount)
	suite.Require().Equal(want, celestiaBalance.Amount)
```

Reply:
Done: [add extra assertion](https://github.com/celestiaorg/celestia-app/pull/1474/commits/d37325c46b1eb19148de19c4764ff0d7a48a37be)